Darren new expand chat

DarrenPoh opened the pull request:
Modified chat expand code

DarrenPoh committed:
no message

DarrenPoh committed:
no message

DarrenPoh committed:
Merge branch 'master' of https://github.com/eddieberklee/android-meet-in-the-middle into darren-new-expand-chat

# Conflicts:
#	app/src/main/java/com/compscieddy/meetinthemiddle/activity/GroupActivity.java
#	app/src/main/java/com/compscieddy/meetinthemiddle/fragment/ChatFragment.java

DarrenPoh committed:
Removed the need for button click.
Made it so the ChatFragment (bottom_section) only has 2 states, expanded (75%) or minimized (30%).
ChatFragment is first called when GroupActivity is opened, so the chat is expanded (need to change chat default size to fix animation)
To minimize the chat, MapDrag is used

eddieberklee commented on app/src/main/java/com/compscieddy/meetinthemiddle/activity/GroupActivity.java:
```diff
@@ -693,7 +722,7 @@ public void run() {
   }
 
-  private void resizeViewPager(boolean isFABClick) {
+  public void resizeViewPager(boolean isFABClick) {
```
#nitpicky isFabClick (rename)

DarrenPoh replied:
i duno who chose that name hahah, id prefer chatExpanded or something

eddieberklee commented on app/src/main/java/com/compscieddy/meetinthemiddle/activity/GroupActivity.java:
```diff
@@ -702,6 +731,20 @@ private void resizeViewPager(boolean isFABClick) {
     if (isFABClick) {
+      resizeAnimation = new ResizeAnimation(
+          mBottomSection,
+          (int) (height * 0.75),
```
Move these out into constants (0.75 and 0.3)

eddieberklee commented on app/src/main/java/com/compscieddy/meetinthemiddle/activity/GroupActivity.java:
```diff
@@ -702,6 +731,20 @@ private void resizeViewPager(boolean isFABClick) {
+          (int) (height * 0.3),
+          (int) (height * 0.75));
+    }
+    resizeAnimation.setDuration(400);
```
Move out into constant

DarrenPoh committed:
Fixed resizing on double scroll
Need to fix first call to onScroll

DarrenPoh committed:
set values as constants.
had to add the TypedValue code because dimens.xml does not naturally allow floats

eddieberklee commented on app/src/main/java/com/compscieddy/meetinthemiddle/activity/GroupActivity.java:
```diff
@@ -693,16 +723,47 @@ public void run() {
     ResizeAnimation resizeAnimation;
+    
+    TypedValue expandedValue = new TypedValue();
+    getResources().getValue(R.dimen.chat_expanded_height, expandedValue, true);
```
nono I said constant. `private final static float EXPANDED_HEIGHT_PERCENTAGE`

eddieberklee commented on app/src/main/java/com/compscieddy/meetinthemiddle/activity/GroupActivity.java:
```diff
@@ -693,16 +723,47 @@ public void run() {
+            (int) (height * expandedHeight),
+            (int) (height * minimizedHeight));
+        isViewPagerCollapsed = !isViewPagerCollapsed;
+        resizeAnimation.setDuration(400);
```
move this 400 into a constant (just like above)

DarrenPoh committed:
set the default height of chat fragment so its in main focus

DarrenPoh committed:
added constants

DarrenPoh committed:
added for message box too

eddieberklee commented on app/src/main/java/com/compscieddy/meetinthemiddle/activity/GroupActivity.java:
```diff
@@ -98,6 +98,11 @@
 
   private final int ANIMATE_CAMERA_REPEAT = 2000;
 
+  private final double CHAT_EXPANDED_HEIGHT = 0.75;
```
should use a float over double, almost all the time

eddieberklee commented on app/src/main/java/com/compscieddy/meetinthemiddle/activity/GroupActivity.java:
```diff
@@ -693,16 +738,38 @@ public void run() {
+            mBottomSection,
+            (int) (height * CHAT_EXPANDED_HEIGHT),
+            (int) (height * CHAT_MINIMIZED_HEIGHT));
+        isViewPagerCollapsed = !isViewPagerCollapsed;
```
You can move these 3 lines out of the if/else statement. Because they are **exactly** identical to the lower lines

DarrenPoh committed:
fixed comments

DarrenPoh committed:
no message

eddieberklee commented on app/src/main/java/com/compscieddy/meetinthemiddle/activity/GroupActivity.java:
```diff
@@ -98,6 +98,11 @@
   private final int ANIMATE_CAMERA_REPEAT = 2000;
 
+  private final float CHAT_EXPANDED_HEIGHT = 0.75f;
+  private final float CHAT_MINIMIZED_HEIGHT = 0.3f;
```
variable name should include something like "FACTOR" since you're using this as  percentage/factor/fraction. It makes reading the code easier for others

eddieberklee commented on app/src/main/java/com/compscieddy/meetinthemiddle/activity/GroupActivity.java:
```diff
@@ -242,6 +247,17 @@ public void afterTextChanged(Editable s) {
+    Display display = getWindowManager().getDefaultDisplay();
+    Point size = new Point();
+    display.getSize(size);
+    int height = size.y;
```
Etils.getScreenHeight()

eddieberklee commented on app/src/main/java/com/compscieddy/meetinthemiddle/fragment/ChatFragment.java:
```diff
@@ -112,7 +114,58 @@ public void onComplete(DatabaseError databaseError, DatabaseReference reference)
+        boolean isChatExpanded = sp.getBoolean("isChatExpanded", true);
+
+        Display display = getActivity().getWindowManager().getDefaultDisplay();
+        Point size = new Point();
```
Delete comments if not needed

eddieberklee commented on app/src/main/java/com/compscieddy/meetinthemiddle/activity/GroupActivity.java:
```diff
@@ -496,6 +512,35 @@ public void onMapClick(LatLng point) {
+
+    ResizeAnimation resizeAnimation;
+
+    SharedPreferences sp = PreferenceManager.getDefaultSharedPreferences(this);
```
Delete comments if not needed

DarrenPoh committed:
Swapped to Etils.getHeight.
Changed expand and minimize factors
